In [1]:
import statsmodels.api as sm
import pandas as pd

Build a regression model.

In [4]:
model_df = pd.read_csv('/Users/parkerharalds/Documents/Lighthouse/Statistical_Modelling_Project/data/model_data')

In [5]:
model_df.head()

empty_slots_mean  free_bikes_mean  total_bikes_mean  distance_max  \
0               7.0             11.0              18.0   1199.915982   
1              13.0              1.0              14.0    951.596216   
2              14.0              0.0              14.0   1138.992659   
3              14.0             11.0              25.0    876.702883   
4               9.0              5.0              14.0   1281.087849   

   distance_mean  distance_min  cafes_count  rating_max  rating_mean  \
0     810.193384    350.540969           18         5.0     3.616667   
1     578.035308     71.274224           12         4.6     3.991667   
2     663.138811    145.760213           20         4.8     3.445000   
3     539.830458     88.697075           12         5.0     3.608333   
4     844.545892    217.292249           16         5.0     3.643750   

   rating_min  review_count_max  review_count_mean  review_count_min  
0         0.0             370.0          57.611111               0.0  
1         3.2             519.0         120.583333               1.0  
2         0.0             103.0          19.050000               0.0  
3         0.0             189.0          31.000000               0.0  
4         0.0             285.0          39.750000               0.0

In [7]:
X = model_df['total_bikes_mean']
y = model_df['cafes_count']

In [ ]:
X = sm.add_constant(X) # adding a constant
lin_reg = sm.OLS(y,X)

model = lin_reg.fit()
print_model = model.summary()


Provide model output and an interpretation of the results. 

In [10]:
print(print_model)

                            OLS Regression Results                            
Dep. Variable:            cafes_count   R-squared:                       0.005
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     1.196
Date:                Fri, 18 Oct 2024   Prob (F-statistic):              0.275
Time:                        18:59:21   Log-Likelihood:                -804.32
No. Observations:                 253   AIC:                             1613.
Df Residuals:                     251   BIC:                             1620.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const               16.5012      1.320  

This model is a terrible predictor of the number of cafes in area based on the total bikes at a given bike station. The r_squared value is 0.005 which is horrible. The p_value is also 0.275 which indicates that we can not reject the null hypothesis and that the total bikes is not statistically significant in determining the number of cafes in an area.

# Stretch

How can you turn the regression model into a classification model?